pkg/controllermanager/controller/seed/seed_extension_check_control_test.go

```diff
@@ -241,7 +329,61 @@ var _ = Describe("ExtensionCheckReconciler", func() {
 		})
 
 		It("should set ExtensionsReady to False (NotAllExtensionsHealthy)", func() {
-			Expect(reconciler.Reconcile(ctx, request)).To(Equal(reconcile.Result{}))
+			Expect(reconciler.Reconcile(ctx, request)).To(Equal(reconcile.Result{RequeueAfter: syncPeriodDuration}))
+		})
+
+		Context("when ExtensionsReady condition threshold is set", func() {
+			BeforeEach(func() {
+				conf := config.SeedExtensionsCheckControllerConfiguration{
+					SyncPeriod: &metav1.Duration{Duration: syncPeriodDuration},
+					ConditionThresholds: []config.ConditionThreshold{{
+						Type:     string(gardencorev1beta1.SeedExtensionsReady),
+						Duration: metav1.Duration{Duration: time.Minute},
+					}},
+				}
+				reconciler = NewExtensionsCheckReconciler(c, conf, fakeClock)
+			})
+
+			It("should set ExtensionsReady to Progressing (NotAllExtensionsHealthy) if it was previously True", func() {
+				existingCondition := gardencorev1beta1.Condition{
+					Type:               "ExtensionsReady",
+					Status:             gardencorev1beta1.ConditionTrue,
+					Reason:             "AllExtensionsReady",
+					Message:            "All extensions installed into the seed cluster are ready and healthy.",
+					LastTransitionTime: metav1.NewTime(fakeClock.Now().Add(-time.Second)),
+					LastUpdateTime:     metav1.NewTime(fakeClock.Now().Add(-time.Second)),
+				}
+				seed.Status.Conditions = []gardencorev1beta1.Condition{existingCondition}
+				Expect(c.Status().Update(ctx, seed)).To(Succeed())
+
+				matchExpectedCondition = beConditionWithStatusReasonAndMessage(gardencorev1beta1.ConditionProgressing, "NotAllExtensionsHealthy", `Some extensions are not healthy: map[foo-2:]`)
+				Expect(reconciler.Reconcile(ctx, request)).To(Equal(reconcile.Result{RequeueAfter: syncPeriodDuration}))
+			})
+
+			It("should set ExtensionsReady to False (NotAllExtensionsHealthy) if it was previously Progressing and threshold has expired", func() {
+				existingCondition := gardencorev1beta1.Condition{
+					Type:               "ExtensionsReady",
+					Status:             gardencorev1beta1.ConditionProgressing,
+					Reason:             "NotAllExtensionsHealthy",
+					Message:            `Some extensions are not valid: map[foo-2:]`,
+					LastTransitionTime: metav1.NewTime(fakeClock.Now().Add(-90 * time.Second)),
+					LastUpdateTime:     metav1.NewTime(fakeClock.Now().Add(-90 * time.Second)),
+				}
+				seed.Status.Conditions = []gardencorev1beta1.Condition{existingCondition}
+				Expect(c.Status().Update(ctx, seed)).To(Succeed())
+
+				matchExpectedCondition = beConditionWithStatusReasonAndMessage(gardencorev1beta1.ConditionFalse, "NotAllExtensionsHealthy", `Some extensions are not healthy: map[foo-2:]`)
+				Expect(reconciler.Reconcile(ctx, request)).To(Equal(reconcile.Result{RequeueAfter: syncPeriodDuration}))
+			})
```

Comment:
Can this be made reusable by putting it into a function? This whole blocks looks very similar to what you were doing above?